### PR TITLE
added template variables and documentation updates

### PR DESCRIPTION
**Pull request by kevineaton**
I added the logic for template variables and updated some of the documentation, including an example in the README. Please let me know if this is closer to what you would like.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,13 +1,12 @@
-module github.com/mailgun/mailgun-go/v3
+module github.com/kevineaton/mailgun-go/v3
```
Comment (thrawn01) [inline go.mod]: This should be `mailgun/` not `kevineaton`. I think `go get` might have rewritten this while you were working on this PR.

File: go.mod [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
 
 require (
 	github.com/davecgh/go-spew v1.1.1
 	github.com/facebookgo/ensure v0.0.0-20160127193407-b4ab57deab51
-	github.com/facebookgo/stack v0.0.0-20160209184415-751773369052 // indirect
-	github.com/facebookgo/subset v0.0.0-20150612182917-8dac2c3c4870 // indirect
 	github.com/go-chi/chi v4.0.0+incompatible
+	github.com/mailgun/mailgun-go/v3 v3.6.1
```
Comment (thrawn01) [inline go.mod]: This should be removed.

File: go.sum [hunk 3]
```diff
@@ -8,6 +8,9 @@ github.com/facebookgo/subset v0.0.0-20150612182917-8dac2c3c4870 h1:E2s37DuLxFhQD
 github.com/facebookgo/subset v0.0.0-20150612182917-8dac2c3c4870/go.mod h1:5tD+neXqOorC30/tWg0LCSkrqj/AR6gu8yY8/fpw1q0=
 github.com/go-chi/chi v4.0.0+incompatible h1:SiLLEDyAkqNnw+T/uDTf3aFB9T4FTrwMpuYrgaRcnW4=
 github.com/go-chi/chi v4.0.0+incompatible/go.mod h1:eB3wogJHnLi3x/kFX2A+IbTBlXxmMeXJVKy9tTv1XzQ=
+github.com/mailgun/mailgun-go v2.0.0+incompatible h1:0FoRHWwMUctnd8KIR3vtZbqdfjpIMxOZgcSa51s8F8o=
```
Comment (thrawn01) [inline go.sum]: this also.

**Comment by kevineaton** [comment]
I made the changes and ran `go mod tidy`, which looks to have the resolutions you asked for.

**Comment by kevineaton** [comment]
The pleasure was mine!

**Pull request merged**